pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -185,6 +201,29 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
 	return nil
 }
 
+func needsUpdate(
+	clusterCRD *apiextensionsv1beta1.CustomResourceDefinition,
+	createError error) bool {
+
+	if errors.IsAlreadyExists(createError) {
+		if clusterCRD.Spec.Validation == nil {
```

Comment:
maybe we should also check if `|| clusterCRD.Labels == nil`?

Reply:
Nil labels are treated as empty, so it's handled in lines 213-217